README.md

```diff
@@ -58,44 +63,49 @@ Verify that the sample application is able to run locally. In order to collect m
+   # Create a service principal with Contributor role to the resource group
+   az ad sp create-for-rbac --name $SERVICEPRINCIPAL --role contributor --scopes /subscriptions/$SUBSCRIPTIONID/resourceGroups/$RESOURCEGROUPNAME --sdk-auth
+   ```
 
-3. Add a secret named `AZURE_SUBSCRIPTION_ID` for the subscription id in the repository settings (Settings -> Secrets -> Add New Secret).
+   **CAUTION:** There is a known bug with git bash. Git Bash will attempt to auto-translate resource IDs. If you encounter this issue, it can be fixed by appending MSYS_NO_PATHCONV=1 to the command. [See this link for further information.](https://github.com/fengzhou-msft/azure-cli/blob/ea149713de505fa0f8ae6bfa5d998e12fc8ff509/doc/use_cli_with_git_bash.md)
 
-4. [Github Actions](https://docs.github.com/en/actions) will be used to automate the workflow and deploy all the necessary resources to Azure. Open the [.github\workflows\devops-starter-workflow.yml](.github\workflows\devops-starter-workflow.yml) and change the environment variables accordingly. Use the `RESOURCEGROUPNAME` and value that you created above.
+2. Use the output of the last command as a secret named `AZURE_CREDENTIALS` in the repository settings (Settings -> Secrets -> New repository secret). Set this as a secret on the repository not on the environment. For more details on configuring the github repository secrets, please see [this guide](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/deploy-github-actions#configure-the-github-secrets)
 
-5. Commit your changes. The commit will trigger the build and deploy jobs within the workflow and will provision all the resources to run the sample application.
+3. [Github Actions](https://docs.github.com/en/actions) will be used to automate the workflow and deploy all the necessary resources to Azure. Open the [.github\workflows\devops-starter-workflow.yml](.github\workflows\devops-starter-workflow.yml) and change the environment variables accordingly. Use the `RESOURCEGROUPNAME` and value that you created above. Be sure to change at a minimum the named variables, such as the `RESOURCEGROUPNAME` and the `REGISTRYNAME`. The `REGISTRYNAME` is a globally unique name and the deployment will fail if this value is not unique. [This resource can guide you with naming conventions.](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules)
+
+4. Commit your changes. The commit will trigger the build and deploy jobs within the workflow and will provision all the resources to run the sample application.
 
 ### Install the Prometheus Server
 
 ```bash
 
 # Define variables
-RESOURCE_GROUP="<insert-resource-group-name-here>"
-CLUSTER_NAME="<insert-cluster-name-here>"
-NAMESPACE="<insert-application-namespace-name-here>"
+RESOURCE_GROUP="MyResourcegroup"
```

Comment:
I debated this. I LOVE using insert. Makes it very clear. I didn't know if you wanted to change it everywhere so that (theoretically) the code could run without changes.

Reply:
Corrected per our convo :)